Tests/PassesTests/PassesTests.swift

```diff
@@ -1,570 +1,523 @@
-        XCTAssertEqual(passJSON["authenticationToken"] as? String, pass.authenticationToken)
-        try XCTAssertEqual(passJSON["serialNumber"] as? String, pass.requireID().uuidString)
-        XCTAssertEqual(passJSON["description"] as? String, passData.title)
-
-        let manifestJSONData = try String(
-            contentsOfFile: passFolder.path.appending("/manifest.json")
-        ).data(using: .utf8)
-        let manifestJSON =
-            try JSONSerialization.jsonObject(with: manifestJSONData!) as! [String: Any]
-        let iconData = try Data(contentsOf: passFolder.appendingPathComponent("/icon.png"))
-        let iconHash = Array(Insecure.SHA1.hash(data: iconData)).hex
-        XCTAssertEqual(manifestJSON["icon.png"] as? String, iconHash)
-        XCTAssertNotNil(manifestJSON["logo.png"])
-        XCTAssertNotNil(manifestJSON["personalizationLogo.png"])
+    @Test func passGeneration() async throws {
+        try await withApp(delegate: delegate) { app, passesService in
+            let passData = PassData(title: "Test Pass")
+            try await passData.create(on: app.db)
+            let pass = try await passData.$pass.get(on: app.db)
+            let data = try await passesService.generatePassContent(for: pass, on: app.db)
+            let passURL = FileManager.default.temporaryDirectory.appendingPathComponent("test.pkpass")
+            try data.write(to: passURL)
+            let passFolder = try Zip.quickUnzipFile(passURL)
+
+            #expect(FileManager.default.fileExists(atPath: passFolder.path.appending("/signature")))
+
+            let passJSONData = try String(contentsOfFile: passFolder.path.appending("/pass.json")).data(using: .utf8)
+            let passJSON = try JSONSerialization.jsonObject(with: passJSONData!) as! [String: Any]
+            #expect(passJSON["authenticationToken"] as? String == pass.authenticationToken)
+            let passID = try pass.requireID().uuidString
+            #expect(passJSON["serialNumber"] as? String == passID)
+            #expect(passJSON["description"] as? String == passData.title)
+
+            let manifestJSONData = try String(contentsOfFile: passFolder.path.appending("/manifest.json")).data(using: .utf8)
+            let manifestJSON = try JSONSerialization.jsonObject(with: manifestJSONData!) as! [String: Any]
+            let iconData = try Data(contentsOf: passFolder.appendingPathComponent("/icon.png"))
+            let iconHash = Array(Insecure.SHA1.hash(data: iconData)).hex
+            #expect(manifestJSON["icon.png"] as? String == iconHash)
+            #expect(manifestJSON["logo.png"] != nil)
+            #expect(manifestJSON["personalizationLogo.png"] != nil)
+        }
     }
 
-    func testPassesGeneration() async throws {
-        let passData1 = PassData(title: "Test Pass 1")
-        try await passData1.create(on: app.db)
-        let pass1 = try await passData1.$pass.get(on: app.db)
+    @Test func passesGeneration() async throws {
```

Comment:
Might be worth making use of the possibility of better names
```suggestion
    @Test("Test generating multiple passes")
    func passesGeneration() async throws {
```